### ISSUE TITLE
Define supported node versions in package.json

### ISSUE THREAD

**Issue by stefanjudis** [loopline-systems/electron-builder#43]
https://docs.npmjs.com/files/package.json#engines

`path.win32.normalize( p );`

Is not supported on every node version - so we should commincate that.

**Comment by Akamaozu** [comment]
Would it be easier to put path as a dependency to be installed from npm?

That way you get around requiring a specific version of node.

**Comment by stefanjudis** [comment]
`path` is a module coning with node itself. So it can't be installed via npm. ;)

**Comment by Akamaozu** [comment]
so even though it's on npm and you can require it doesn't mean you can use it?

Interesting.

https://www.npmjs.com/package/path

**Comment by stefanjudis** [comment]
Whooot? 

This doesn't make sense to me at all... :D 

Never the less I'd stick to the "real" one coming with node and simply restricting node versions, what do you think?

**Comment by Akamaozu** [comment]
If the only reason we need a specific version of node is because of the version of path it comes with and we can simply require a newer version of path, I think it's worth it.

If you're concerned, we can lock path version down ...

**Comment by stefanjudis** [comment]
Let me do some research there. :)

I see the point - but extracting native modules and mixing node-versions and its native module versions feels really bad to me.

Will have a detailed look at the weekend.

**Comment by stefanjudis** [comment]
@Akamaozu 

Hey hey, I checked again and actually the `path` module on npm is already behind the current natively shipped `path` module.

Especially since io.js was merged back into node it's moving really really fast.

The npm `path` is missing one commit right now.

https://github.com/jinder/path/blob/master/path.js#L425-L427
https://github.com/nodejs/node/blob/ac2bce0b0c91ba9fc593d8c53c68b42d3c1e4954/lib/path.js#L410-L414

So it might be that we rely on something which we don't need to and which is another dependency. And the who knows if the author is really adding the stuff, that goes into `path` in node itself.

I'm running mostly on a recent node version - so maybe that's why I don't see a problem in restrictring `~v0.12.0` and don't like using a copy of the original.
